src/constructor/load.py

```diff
@@ -0,0 +1,221 @@
+    
+    Args:
+        checked_state_dict: Checked state dict.
+        model_keys: Model state dict keys.
+        checked_module_name: The name of the module to be checked.
+
+    Raises:
+        ValueError: If checked_state_dict keys do not completely intersect with model_keys for current 
+            checked_module_name.
+    """
+    if checked_module_name != '':
+        module_straighten_keys = set(get_straighten_keys(checked_module_name, model_keys))
+    else:
+        module_straighten_keys = set(model_keys)
+
+    checked_keys = set(checked_state_dict.keys())
+    intersection_keys = checked_keys.intersection(module_straighten_keys)
+
+    if len(checked_keys) != len(intersection_keys):
+        missing_keys = module_straighten_keys - checked_keys
+        extra_keys = checked_keys - module_straighten_keys
+        module_part = checked_module_name if checked_module_name != '' else 'all the model'
+        raise ValueError(f'Load checkpoint module, check_state_dict_keys function. found a mismatch between loaded '
+                         f'keys and model keys in {module_part}. Missing keys = {missing_keys}. '
+                         f'Extra keys = {extra_keys}.')
+
+    
+def sort_state_dict_by_depth(override_name2state_dict: Dict[str, str]) -> List[List[OrderedDict[str, torch.Tensor]]]:
+    """Generate sorted by depth list of state dict list with the current depth.
+    Where depth is calculated as the number of dots in the dictionary key.
+
+    Args:
+        override_name2state_dict: Dicts of module key to state dict, which should override base checkpoint.
+
+    Returns:
+        sorted_state_dicts: Sorted by depth list of state dicts list. Where the positional index means the depth 
+            for each state dicts at this position.
+    """
+    max_depth = 0
+    sorted_state_dicts = [[]] * len(override_name2state_dict)
+
+    for override_key, override_state_dict in override_name2state_dict.items():
+        depth = len(override_key.split('.')) - 1
+        sorted_state_dicts[depth].append(override_state_dict)
+        max_depth = depth if depth > max_depth else max_depth
+
+    sorted_state_dicts = sorted_state_dicts[: max_depth + 1]
+    return sorted_state_dicts
+
+
+def get_state_dict_with_prefix(prefix: str, 
+                               state_dict: OrderedDict[str, torch.Tensor]) -> OrderedDict[str, torch.Tensor]:
+    """Generate state dict with prefixed keys. If input state dict keys startswith prefix then no prefix added.
+
+    Args:
+        prefix: Prefix for state dict keys that should be added, then state dict keys not begin with prefix.
+        state_dict: State dict whose keys should be prefixed.
+
+    Returns:
+        state_dict_with_prefix: Prefixed state dict.
+    """
+    state_dict_with_prefix = OrderedDict()
+    # remove spaces and dots
+    prefix = prefix.strip(' .')
+    for key, value in state_dict.items():
+        if not key.startswith(prefix):
```

Comment:
What if a key does itself contain a substring starting with a prefix? Example:
```
override_state_dict = {
  'linear': {
    'linear_layer': torch.Tensor(...)
  }
}
```
The absolute key must be: `linear.linear_layer`, but your algorithm will make it just `linear_layer`. I suggest leaving the job of putting the correct prefixes to a user because it's unclear where they start. So, just always put prefix before the key (and I guess there's no need for a special method)